packages/zowe-explorer/src/trees/dataset/DatasetTree.ts

```diff
@@ -166,7 +181,9 @@ export class DatasetTree extends ZoweTreeProvider<IZoweDatasetTreeNode> implemen
                 }
             } catch (err) {
                 //file might already exist. Ignore the error and try to write it to lpar
-                if (err.errorCode.toString() === "404" || err.errorCode.toString() === "500") {
+                // only bail out on a real 404/500 – guard against undefined
+                const code = err?.errorCode?.toString?.();
```

Comment:
Coverity won't like this, since we assume `err` is defined on line 187, but handle the case of it being undefined here.

We should be consistent - since `err` should always be defined in a catch block, and JS objects should always define a `toString` implementation, I think we can do this:
```suggestion
                const code = err.errorCode?.toString();
```